swiftpkg/internal/bzl_selects.bzl

```diff
@@ -181,18 +182,33 @@ def _to_starlark(values, kind_handlers = {}):
     if sets.length(no_condition_results) > 0:
         expr_members.append(sets.to_list(no_condition_results))
     for (kind, select_dict) in selects_by_kind.items():
-        if len(expr_members) > 0:
-            expr_members.append(scg.new_op("+"))
-        sorted_keys = sorted(select_dict.keys())
-        new_select_dict = {
-            k: sets.to_list(select_dict[k])
-            for k in sorted_keys
-        }
         kind_handler = kind_handlers.get(kind, _noop_kind_handler)
-        if kind_handler.default != None:
-            new_select_dict[_bazel_select_default_condition] = kind_handler.default
-        select_fn = scg.new_fn_call("select", new_select_dict)
-        expr_members.append(select_fn)
+        sorted_keys = sorted(select_dict.keys())
+
+        if mutually_inclusive:
+            # Generate multiple select expressions for each condition.
+            for k in sorted_keys:
+                if len(expr_members) > 0:
+                    expr_members.append(scg.new_op("+"))
+                new_dict = {
+                    k: sets.to_list(select_dict[k]),
+                }
+                if kind_handler.default != None:
+                    new_dict[_bazel_select_default_condition] = kind_handler.default
+                select_fn = scg.new_fn_call("select", new_dict)
+                expr_members.append(select_fn)
+        else:
+            # Combine all conditions of the same kind into one select expression.
+            if len(expr_members) > 0:
+                expr_members.append(scg.new_op("+"))
+            new_select_dict = {
+                k: sets.to_list(select_dict[k])
+                for k in sorted_keys
+            }
+            if kind_handler.default != None:
+                new_select_dict[_bazel_select_default_condition] = kind_handler.default
+            select_fn = scg.new_fn_call("select", new_select_dict)
+            expr_members.append(select_fn)
```

Comment:
Maybe we should move this to an internal function so we can reuse the logic in both branches at seems similar across the if/else. Then the only difference would be the for loop

Reply:
Thanks a lot for the idea!  I had to rely on `expr_members` being passed by reference to extract more logic and I feel like it makes the code a little bit harder to read, but I'm a starlark novice, so maybe I'm just not used to this behaviour being implicit. What do you think?

Reply:
Coming from swift it does feel a lil weird but I think that's pretty common in Starlark/python so this is looking better IMO, thank you!